--- gcc/tree.h.orig	2024-03-26 23:10:21.115634000 -0500
+++ gcc/tree.h	2024-03-26 23:32:30.584953000 -0500
@@ -1299,7 +1299,6 @@
 
 extern void protected_set_expr_location (tree, location_t);
 extern void protected_set_expr_location_if_unset (tree, location_t);
-ATTRIBUTE_WARN_UNUSED_RESULT
 extern tree protected_set_expr_location_unshare (tree, location_t);
 
 WARN_UNUSED_RESULT extern tree maybe_wrap_with_location (tree, location_t);
