diff -ru Python-2.7.18/Objects/unicodeobject.c Python-2.7.18.new/Objects/unicodeobject.c
--- Python-2.7.18/Objects/unicodeobject.c	2020-04-19 17:13:39.000000000 -0400
+++ Python-2.7.18.new/Objects/unicodeobject.c	2020-11-22 20:37:50.188878382 -0500
@@ -2133,6 +2133,13 @@
         continue;
 
       utf8Error:
+
+        if ((!strcmp(errors, "strict")) || (!strcmp(errors, "surrogateescape"))) {
+            *p++ = (Py_UNICODE)(0xDC00 + (s[0] & 0xff));
+            s += 1;
+            continue;
+        }
+
         outpos = p-PyUnicode_AS_UNICODE(unicode);
         if (unicode_decode_call_errorhandler(
                 errors, &errorHandler,
@@ -2140,6 +2147,7 @@
                 starts, size, &startinpos, &endinpos, &exc, &s,
                 &unicode, &outpos, &p))
             goto onError;
+
     }
     if (consumed)
         *consumed = s-starts;
@@ -2228,6 +2236,13 @@
                     }
                     /* Fall through: handles isolated high surrogates */
                 }
+
+                /* Handle Python 3-style surrogate escapes. */
+                if (0xDC80 <= ch && ch <= 0xDCFF) {
+                    *p++ = (char)(ch & 0xff);
+                    continue;
+                }
+
                 *p++ = (char)(0xe0 | (ch >> 12));
                 *p++ = (char)(0x80 | ((ch >> 6) & 0x3f));
                 *p++ = (char)(0x80 | (ch & 0x3f));
diff -ru Python-2.7.18/Python/pythonrun.c Python-2.7.18.new/Python/pythonrun.c
--- Python-2.7.18/Python/pythonrun.c	2020-04-19 17:13:39.000000000 -0400
+++ Python-2.7.18.new/Python/pythonrun.c	2020-11-22 20:13:58.741719276 -0500
@@ -290,6 +290,13 @@
     if (!Py_NoSiteFlag)
         initsite(); /* Module site */
 
+
+    icodeset = codeset = "utf-8";
+    errors = "strict";
+    free_codeset = 0;
+    overridden = 1;
+
+# if 0
     if ((p = Py_GETENV("PYTHONIOENCODING")) && *p != '\0') {
         p = icodeset = codeset = strdup(p);
         free_codeset = 1;
@@ -300,6 +307,7 @@
         }
         overridden = 1;
     }
+#endif
 
 #if defined(Py_USING_UNICODE) && defined(HAVE_LANGINFO_H) && defined(CODESET)
     /* On Unix, set the file system encoding according to the
