diff -ru Python-2.7.18/Python/pythonrun.c Python-2.7.18.new/Python/pythonrun.c
--- Python-2.7.18/Python/pythonrun.c	2020-04-19 17:13:39.000000000 -0400
+++ Python-2.7.18.new/Python/pythonrun.c	2021-09-29 19:24:32.051745167 -0400
@@ -311,6 +311,11 @@
     if (!overridden || !Py_FileSystemDefaultEncoding) {
         saved_locale = strdup(setlocale(LC_CTYPE, NULL));
         setlocale(LC_CTYPE, "");
+
+        if (!strcmp(setlocale(LC_CTYPE, NULL), "C")) {
+            setlocale(LC_CTYPE, "C.UTF-8");
+        }
+
         loc_codeset = nl_langinfo(CODESET);
         if (loc_codeset && *loc_codeset) {
             PyObject *enc = PyCodec_Encoder(loc_codeset);
@@ -2043,4 +2048,3 @@
 #ifdef __cplusplus
 }
 #endif
-
