diff -r -u Python-2.7.17/setup.py Python-2.7.17.new/setup.py
--- Python-2.7.17/setup.py	2019-10-19 14:38:44.000000000 -0400
+++ Python-2.7.17.new/setup.py	2019-12-01 23:58:09.268652755 -0500
@@ -422,6 +422,9 @@
             self.failed.append(ext.name)
 
     def add_multiarch_paths(self):
+        # renpy-build - disable this, since everything is a cross-compile.
+        return
+
         # Debian/Ubuntu multiarch support.
         # https://wiki.ubuntu.com/MultiarchSpec
         cc = sysconfig.get_config_var('CC')
