commit 60889d78308534222071d73dc38f2a43878b3d9b
Author: Tom Rothamel <tom@rothamel.us>
Date:   Sun Sep 22 04:11:08 2024 -0400

    Disable the system call on iOS.

diff --git a/Modules/posixmodule.c b/Modules/posixmodule.c
index 6fb2fc8..7dcb402 100644
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -4978,6 +4978,7 @@ os_rmdir_impl(PyObject *module, path_t *path, int dir_fd)
     Py_RETURN_NONE;
 }
 
+#undef HAVE_SYSTEM
 
 #ifdef HAVE_SYSTEM
 #ifdef MS_WINDOWS
